components/mediation-monitor/mediation-data-publisher/org.wso2.carbon.prometheus.publisher/src/main/java/org/wso2/carbon/prometheus/publisher/internal/MetricServiceComponent.java

```diff
@@ -0,0 +1,111 @@
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.prometheus.publisher.internal;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.osgi.service.component.ComponentContext;
+import org.wso2.carbon.base.ServerConfiguration;
+import org.wso2.carbon.prometheus.publisher.service.MetricService;
+import org.wso2.carbon.prometheus.publisher.util.PrometheusPublisherConstants;
+import org.wso2.msf4j.MicroservicesRunner;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @scr.component name="org.wso2.carbon.prometheus.publisher" immediate="true"
+ **/
+
+public class MetricServiceComponent {
+
+    private static final Log LOGGER = LogFactory.getLog(MetricServiceComponent.class);
+
+    private boolean isPublishingEnabled = false;
+    private int port = 1289;
+
+    /**
+     * Activates the component, reads config and starts the metric service
+     *
+     * @param context This is required for the component to be activated as an OSGi component
+     */
+    protected void activate(ComponentContext context) {
+
+        readConfig();
+        if (isPublishingEnabled) {
+            LOGGER.info("Metrics published to Prometheus on port: " + port);
+            new MicroservicesRunner(port)
+                    .deploy(new MetricService())
+                    .start();
+        }
+    }
+
+    /**
+     * Reads config data from carbon.xml
+     */
+    private void readConfig() {
+
+        ServerConfiguration serverConf = ServerConfiguration.getInstance();
+        isPublishingEnabled =
+                Boolean.parseBoolean(
+                        serverConf.getFirstProperty(PrometheusPublisherConstants.PROMETHEUS_PUBLISHING_ENABLED));
+
+        if (isPublishingEnabled) {
+            String portValue = serverConf.getFirstProperty(PrometheusPublisherConstants.PUBLISHING_PORT);
+            if (isValidPort(portValue)) {
+                port = Integer.parseInt(portValue);
+            } else {
+                port = 1289;
+                if (LOGGER.isDebugEnabled()) {
+                    LOGGER.debug("Invalid port configuration. Using default port: 1289");
+                }
+            }
+        }
+    }
+
+    private boolean configIsValid(String value) {
+
+        return value != null && !value.isEmpty();
+    }
+
+    private boolean isValidPort(String port) {
+
+        if (!configIsValid(port))
+            return false;
```

Comment:
We need to have curly braces even though we only have one statement according to the code style. Shall we fix it?